db4-storage/src/segments/graph_prop/mod.rs

```diff
@@ -0,0 +1,90 @@
+pub mod entry;
+pub mod segment;
+
+use crate::{
+    api::graph_props::GraphPropSegmentOps,
+    error::StorageError,
+    persist::strategy::Config,
+    segments::graph_prop::{entry::MemGraphPropEntry, segment::MemGraphPropSegment},
+};
+use parking_lot::{RwLock, RwLockReadGuard, RwLockWriteGuard};
+use raphtory_api::core::entities::properties::meta::Meta;
+use std::{
+    path::Path,
+    sync::{
+        Arc,
+        atomic::{AtomicBool, AtomicUsize, Ordering},
+    },
+};
+
+/// `GraphPropSegmentView` manages graph temporal properties and graph metadata
+/// (constant properties). Reads / writes are always served from the in-memory segment.
+#[derive(Debug)]
+pub struct GraphPropSegmentView<P: Config> {
+    /// In-memory segment that contains the latest graph properties
+    /// and graph metadata writes.
+    head: Arc<RwLock<MemGraphPropSegment>>,
+
+    /// Estimated size of the segment in bytes.
+    est_size: AtomicUsize,
+
+    is_dirty: AtomicBool,
+
+    _persistent: P,
+}
+
+impl<P: Config> GraphPropSegmentOps for GraphPropSegmentView<P> {
+    type Extension = P;
+
+    type Entry<'a> = MemGraphPropEntry<'a>;
+
+    fn new(meta: Arc<Meta>, _path: Option<&Path>, ext: Self::Extension) -> Self {
+        Self {
+            head: Arc::new(RwLock::new(MemGraphPropSegment::new_with_meta(meta))),
+            est_size: AtomicUsize::new(0),
+            is_dirty: AtomicBool::new(false),
+            _persistent: ext,
+        }
+    }
+
+    fn load(
+        _meta: Arc<Meta>,
+        _path: impl AsRef<Path>,
+        _ext: Self::Extension,
+    ) -> Result<Self, StorageError> {
+        todo!()
```

Comment:
shouldn't this just return an error (I am assuming this is never going to get implemented?)